server/src/main/java/org/opensearch/search/aggregations/bucket/terms/ParsedTerms.java

```diff
@@ -141,7 +141,7 @@ static <B extends ParsedBucket> B parseTermsBucketXContent(
             while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
                 if (token == XContentParser.Token.FIELD_NAME) {
                     currentFieldName = parser.currentName();
-                } else if (token.isValue()) {
+                } else if (token.isValue() || token == XContentParser.Token.START_ARRAY) {
```

Comment:
This change does not look right: if token is an array, we should handle it separately, the `parser.text()` for array would throw `IllegalStateException`

Reply:
Done. Refactor the code.